### PR TITLE
[improve][cli]Introduce Regex and File Input Parameters for Enhanced Topic Deletion Command

### PR DESCRIPTION
**Pull request by liangyepianzhou**
PIP: https://github.com/apache/pulsar/issues/20225
### Motivation

The current implementation of the `DeleteCmd` class in the Pulsar CLI does not provide the flexibility to delete multiple topics at once using a regex or a file containing a list of topics. This can be inconvenient and time-consuming in environments where there are many topics that need to be deleted. 

### Modifications

To address this, we have added three new parameters to the `DeleteCmd` class:

1. `-r` or `regex`: This parameter allows the user to specify a regex to match multiple topics for deletion.
2. `--from-file`: This parameter allows the user to read a list of topics from a file for deletion.

Additionally, we have written unit tests to ensure the correct behavior of these new features. These tests cover the deletion of a single topic, multiple topics via regex, and multiple topics from a file.

### Verifying this change

- [ ] Make sure that the change passes the CI checks.

*(Please pick either of the following options)*

This change is a trivial rework / code cleanup without any test coverage.

*(or)*

This change is already covered by existing tests, such as *(please describe tests)*.

*(or)*

This change added tests and can be verified as follows:

*(example:)*
  - *Added integration tests for end-to-end deployment with large payloads (10MB)*
  - *Extended integration test for recovery after broker failure*

### Does this pull request potentially affect one of the following parts:

*If the box was checked, please highlight the changes*

- [ ] Dependencies (add or upgrade a dependency)
- [ ] The public API
- [ ] The schema
- [ ] The default values of configurations
- [ ] The threading model
- [ ] The binary protocol
- [ ] The REST endpoints
- [ ] The admin CLI options
- [ ] The metrics
- [ ] Anything that affects deployment

### Documentation

- [ ] `doc` 
- [ ] `doc-required` 
- [x] `doc-not-needed` 
- [ ] `doc-complete` 

### Matching PR in forked repository

PR in forked repository:

### CODE REVIEW

File: pulsar-client-tools/src/main/java/org/apache/pulsar/admin/cli/CmdTopics.java [hunk 1]
```diff
@@ -726,10 +734,38 @@ private class DeleteCmd extends CliCommand {
...
-            getTopics().delete(topic, force);
+        void run() throws PulsarAdminException, IOException {
+            if (readFromFile && regex) {
+                throw new ParameterException("Could not apply regex when read topics from file.");
+            }
+            if (readFromFile) {
+                String path = checkArgument(params);
+                List<String> topicsFromFile = Files.readAllLines(Path.of(path));
+                for (String t : topicsFromFile) {
+                    getTopics().delete(t);
```
Comment (Technoboy-) [inline pulsar-client-tools/src/main/java/org/apache/pulsar/admin/cli/CmdTopics.java:754]: Why not `force`?

File: pulsar-client-tools/src/main/java/org/apache/pulsar/admin/cli/CmdTopics.java [hunk 2]
```diff
@@ -714,9 +719,12 @@ void run() throws Exception {
...
             + "data of the topic instead of delete cmd if users continue to use this topic later.")
-    private class DeleteCmd extends CliCommand {
-        @Parameter(description = "persistent://tenant/namespace/topic", required = true)
-        private java.util.List<String> params;
+    protected class DeleteCmd extends CliCommand {
+        @Parameter(description = "Provide either a single topic in the format 'persistent://tenant/namespace/topic', "
+                + "or a path to a file containing a list of topics, e.g., 'path://resources/topics.txt'. "
+                + "This parameter is required.",
+                required = true)
+        java.util.List<String> params;
```
[the remark this answers is not in the document]
Comment (Technoboy-) [reply pulsar-client-tools/src/main/java/org/apache/pulsar/admin/cli/CmdTopics.java:727]: Ah, i see the test and get the format.

File: pulsar-client-tools/src/main/java/org/apache/pulsar/admin/cli/CmdTopics.java [hunk 3]
```diff
@@ -105,7 +110,7 @@ public CmdTopics(Supplier<PulsarAdmin> admin) {
         jcommander.addCommand("lookup", new Lookup());
         jcommander.addCommand("partitioned-lookup", partitionedLookup);
         jcommander.addCommand("bundle-range", new GetBundleRange());
-        jcommander.addCommand("delete", new DeleteCmd());
+        jcommander.addCommand("delete", deleteCmd);
```
[the remark this answers is not in the document]
Comment (Technoboy-) [reply pulsar-client-tools/src/main/java/org/apache/pulsar/admin/cli/CmdTopics.java:113]: seems no tests using it right now.

File: pulsar-client-tools/src/main/java/org/apache/pulsar/admin/cli/CmdTopics.java [hunk 4]
```diff
@@ -726,10 +734,38 @@ private class DeleteCmd extends CliCommand {
                 + "but the parameter is invalid and the schema is always deleted", hidden = true)
         private boolean deleteSchema = false;
 
+        @Parameter(names = {"-r", "regex"},
+                description = "Use a regex expression to match multiple topics for deletion.")
+        boolean regex = false;
+
+        @Parameter(names = {"--from-file"}, description = "Read a list of topics from a file for deletion.")
```
[the remark this answers is not in the document]
Comment (liangyepianzhou) [reply pulsar-client-tools/src/main/java/org/apache/pulsar/admin/cli/CmdTopics.java:741]: Should it be consistent with the proposal?

File: pulsar-client-tools/src/main/java/org/apache/pulsar/admin/cli/CmdTopics.java (hunk 3 above)
[the remark this answers is not in the document]
Comment (liangyepianzhou) [reply pulsar-client-tools/src/main/java/org/apache/pulsar/admin/cli/CmdTopics.java:113]: It is used here. In this way, we can set admin without overwriting `getAdmin()`
```
deleteCmd = spy(cmdTopics.getDeleteCmd());
```

**Review by Technoboy- — APPROVED** [review]

File: pulsar-client-tools/src/main/java/org/apache/pulsar/admin/cli/CmdTopics.java [hunk 5]
```diff
@@ -726,10 +734,38 @@ private class DeleteCmd extends CliCommand {
...
-            getTopics().delete(topic, force);
+        void run() throws PulsarAdminException, IOException {
+            if (readFromFile && regex) {
+                throw new ParameterException("Could not apply regex when read topics from file.");
+            }
+            if (readFromFile) {
+                String path = checkArgument(params);
+                List<String> topicsFromFile = Files.readAllLines(Path.of(path));
+                for (String t : topicsFromFile) {
+                    getTopics().delete(t, force);
```
Comment (poorbarcode) [inline pulsar-client-tools/src/main/java/org/apache/pulsar/admin/cli/CmdTopics.java:754]: If one topic has been deleted before, what will happen here?

File: pulsar-client-tools/src/test/java/org/apache/pulsar/admin/cli/TestCmdTopics.java [hunk 6]
```diff
@@ -160,4 +168,59 @@ public void testPartitionedLookupSortByBroker() throws Exception {
         partitionedLookup.sortByBroker = true;
         verify(mockLookup).lookupPartitionedTopic(eq(topic.toString()));
     }
+    @Test
+    public void testRunDeleteSingleTopic() throws PulsarAdminException, IOException {
```
Comment (poorbarcode) [inline pulsar-client-tools/src/test/java/org/apache/pulsar/admin/cli/TestCmdTopics.java:172]: Could you also add a test that some topics could not be deleted successfully?

**Review by poorbarcode — APPROVED** [review]

**Pull request merged**